packages/cli/src/index.js

```diff
@@ -83,14 +84,52 @@ const loadDotEnvDefaultsMiddleware = () => {
   })
 }
 
+const updateCheckerMiddleware = (argv) => {
+  if (argv._[0] === 'update' || argv._[0] === 'upgrade') {
+    return
+  }
+  if (updateCommand.isUpgradeAvailable()) {
+    process.on('exit', () => {
+      console.log(updateCommand.upgradeAvailableMessage())
+    })
+  }
+  if (updateCommand.isUpdateCheckDue()) {
+    const backgroundUpdateBaseLogPath = path.join(
+      getPaths().base || '/tmp',
+      '.redwood',
+      'update'
+    )
+    const out = fs.openSync(
+      path.join(backgroundUpdateBaseLogPath, 'out.log'),
+      'a'
+    )
+    const err = fs.openSync(
+      path.join(backgroundUpdateBaseLogPath, 'err.log'),
+      'a'
+    )
+    const child = spawn('yarn', ['rw', 'update', '--silent'], {
+      detached: true,
+      stdio: ['ignore', out, err],
+      cwd: getPaths().base,
+      env: { ...process.env },
+    })
+    child.unref()
+  }
+}
+
+const yargsMiddleware = [
+  getCwdMiddleware,
+  loadDotEnvDefaultsMiddleware,
+  telemetryMiddleware,
+]
+if (!process.env.REDWOOD_DISABLE_BACKGROUND_UPDATES) {
+  yargsMiddleware.push(updateCheckerMiddleware)
+}
+
 // eslint-disable-next-line no-unused-expressions
 yargs
   .scriptName('rw')
-  .middleware([
-    getCwdMiddleware,
-    loadDotEnvDefaultsMiddleware,
-    telemetryMiddleware,
-  ])
+  .middleware(yargsMiddleware)
```

Comment:
```suggestion
  .middleware([
    getCwdMiddleware,
    loadDotEnvDefaultsMiddleware,
    telemetryMiddleware,
    updateCheckerMiddleware && !process.env.REDWOOD_DISABLE_BACKGROUND_UPDATES
  ].filter(Boolean))
```

This isn't any better/worse than what you had. It's just more inline with how we do similar things in other places in the codebase

Reply:
Sorry I might have been too willing to accept that without actually testing it. Using this change it would appear that `updateCheckerMiddleware && !process.env.REDWOOD_DISABLE_BACKGROUND_UPDATES` doesn't return a function but returns a boolean. The middleware requires a function.

Reply:
My bad. Should have switched those around. `!proc... && updateChe...`